### ISSUE TITLE
creating new repository for libmatewnck

### ISSUE THREAD

**Issue by raveit65** [mate-desktop/libmatewnck#1]
Forking libwnck again, for several reason.
- Workspace switcher is already broken with libwnck https://github.com/mate-desktop/mate-panel/issues/1230
- libwnck dev do not work on PR from mate devs since 4 years. https://gitlab.gnome.org/GNOME/libwnck/-/merge_requests/10
- and their roadmap looks like more breakages. https://gitlab.gnome.org/GNOME/libwnck/-/issues/150
- last but not least working with libwnck , gnome-flashback maintainer is very problematic. I' am really tired of him

So, re-forking libwnck will solve a lot of problems, ihmo.

**Comment by raveit65** [comment]
Cloning is done. https://github.com/mate-desktop/libmatewnck
Should i reset master branch to a specific commit?
ie.
[Release: 43.0](https://github.com/mate-desktop/libmatewnck/commit/6147abd0c1fe35b96e1fbb63fa94d23f91fb2934) 
Tested and running in fedora stable branches (exept reverted workspace-switcher commit)

**Comment by vkareh** [comment]
@raveit65 no, let's track `upstream/master` for a bit and issue pull request for features so that we can track on our own here.

**Comment by raveit65** [comment]
Ok, than we work on a release branch, make sense.
Hopefully rebasing from uptream/master will work because of different repository name.

**Comment by zhuyaliang** [comment]
The advantage is that we can decide our own development path, and we can develop and write our own code.
The downside is that this will increase our maintenance costs
I may not make any suggestions in the short term, I need to familiarize myself with the wnck code first

**Comment by lukefromdc** [comment]
Two test builds (master and cairo-surfaces) just finished fine

To use this now, it should be possible to simply install in the place of upstream libwnck as the filenames are the same. Unless we get buy-in from the other projects (including all WM's commonly used with MATE not just marco and compiz we will need to keep these the same. It should then work as a drop-in replacement unless a project chooses to use a symbol added upstream

**Comment by raveit65** [comment]
2 commits needs to be reverted from libwnck-0.43.
This fixes workspace-switcher:
https://src.fedoraproject.org/rpms/libwnck3/c/e20600dcd0bd1d30948194ce2212cf6e315a038f?branch=rawhide
In our repo: https://github.com/mate-desktop/libmatewnck/commit/3456b747b6381f17d48629dd8fdd4d511e739b10
And for a ubuntu bug:
https://src.fedoraproject.org/rpms/libwnck3/c/f0248c33c5188df7859c3ab0644bb4d78eb0da1a?branch=rawhide
In our repo:
https://github.com/mate-desktop/libmatewnck/commit/07694559cc0c65ce1cca9ac33b165cef84c34d5e

But let me first setup CI.

We need to rename libs and some path to void conflicts with libwnck.
Otherwise it won't be possible to use it for a distribution.
Both packages needs to be installable side by side.

**Comment by raveit65** [comment]
Libwnck use this dirs:
```
%files -f %{source_name}-3.0.lang
 %license COPYING
 %doc AUTHORS README NEWS
 %{_libdir}/%{source_name}-3.so.0*
 %{_bindir}/wnck-urgency-monitor
 %{_libdir}/girepository-1.0/Wnck-3.0.typelib
   
 %files devel
 %{_bindir}/wnckprop
 %{_libdir}/%{source_name}-3.so
 %{_libdir}/pkgconfig/*
 %{_includedir}/%{source_name}-3.0/
 %{_datadir}/gir-1.0/Wnck-3.0.gir
 %doc %{_datadir}/gtk-do
```

**Comment by raveit65** [comment]
I added all our teams to the repository.
You should have the same rights like for the other repos

**Comment by raveit65** [comment]
I created a mate-master branch and setup it as default branch.
This is our main working branch where we can do pull requests to update libmatewnck code and make our releases from.
It's necessary not to commit on master to avoid that we are not in sync with upstream master to track future commits from them with .
An example sync with upstream can be done with:
```
git remote add upstream git://github.com/user/repo.git
git pull upstream master
git push origin
```
The future upstream commits can be cherry-picked in mate-master branch if we want.

**Comment by raveit65** [comment]
@vkareh 
I am not sure about reverting https://github.com/mate-desktop/libmatewnck/commit/07694559cc0c65ce1cca9ac33b165cef84c34d5e
I reverted this in fedora rpm to fix this bug https://gitlab.gnome.org/GNOME/libwnck/-/issues/154#note_1563127
But after 0.43 release are a few commits related to xutils. Maybe those are for fixing the issue?
